### PR TITLE
ISPN-14814 ServerEventLogger blocks when creating a cache

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-14814

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/logging/events/ServerEventLogger.java [hunk 1]
```diff
@@ -72,7 +69,23 @@ void textLog(EventLogLevel level, EventLogCategory category, String message) {
    }
 
    void eventLog(ServerEventImpl event) {
-      getEventCache().putAsync(Util.threadLocalRandomUUID(), event)
+      if (eventCache == null) {
+         if (!cacheManager.isRunning(EVENT_LOG_CACHE)) {
+            // Cache is not running yet, we can't block this thread so offload blocking thread to log the message
+            blockingManager.runBlocking(() -> {
+               eventCache = cacheManager.getCache(EVENT_LOG_CACHE);
```
Comment (jabolina) [inline server/runtime/src/main/java/org/infinispan/server/logging/events/ServerEventLogger.java:76]: Do we need any visibility change for `eventCache`?

Comment (wburns) [reply server/runtime/src/main/java/org/infinispan/server/logging/events/ServerEventLogger.java:76]: Naw, at worst case if a thread can't see it, it just calls getCache again.

Comment (wburns) [reply server/runtime/src/main/java/org/infinispan/server/logging/events/ServerEventLogger.java:76]: And it is using putAsync, so ordering was never guaranteed.

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Integrated into main, thanks @wburns!

**Pull request merged**